pkg/binding/annotationmapper.go

```diff
@@ -45,23 +46,32 @@ func NewDefinitionBuilder(annotationName string, annotationValue string, configM
 	}
 }
 
+var skip = errors.New("skip")
+
 func (m *annotationBackedDefinitionBuilder) outputName() (string, error) {
-	// bail out in the case the annotation name doesn't start with "service.binding"
-	if m.name != AnnotationPrefix && !strings.HasPrefix(m.name, AnnotationPrefix+"/") {
```

Comment:
we were accepting following annotations so far:

* `service.binding: value`
* `service.binding/xx: value`

Now, this is not the case anymore. We should revert back to that logic. We should only ignore `service.binding/provisioned-service` annotation

Reply:
We also need to explicitly ignore non-service binding annotations here (such as `foo: bar`).  Previously, we were throwing an error when we found these annotations, but since we were also ignoring the errors that `annotationBackedDefinitionBuilder.Build` would throw, the result was effectively the same.

Since we're no longer ignoring these errors, we need to make these annotations not return errors (since they're not errors that we should stop binding for).  I think we should only return errors in the case that we need to either retry or stop processing the binding request, and an annotation not under the service binding namespace isn't such an error.